### PR TITLE
[libzenohc] Update to 1.0.0

### PR DESCRIPTION
**Pull request by n0wis**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: libzenohc
* Version: v1.0.0

@staticfloat please review and merge.

### CODE REVIEW

File: L/libzenohc/build_tarballs.jl [hunk 1]
```diff
@@ -3,29 +3,32 @@
...
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/eclipse-zenoh/zenoh-c.git", "a0bacbb65b11e1e5ecedb3b3888beba78bf227cd")
+    GitSource("https://github.com/eclipse-zenoh/zenoh-c.git", "bca57bb6e5a55421cc6ef20ddae5fd3cbb58c30c")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
+cd $WORKSPACE/srcdir
```
Comment (giordano) [inline L/libzenohc/build_tarballs.jl:15]:
```suggestion
```

File: L/libzenohc/build_tarballs.jl [hunk 2]
```diff
@@ -3,29 +3,32 @@
...
+cmake -B build -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DZENOHC_CUSTOM_TARGET=$CARGO_BUILD_TARGET
 cmake --build build --parallel ${nproc}
-cmake --install build
+cmake --build build --target install
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
+    Platform("aarch64", "macos"; ),
```
Comment (giordano) [inline L/libzenohc/build_tarballs.jl:25]: This changed the order of the platforms, which (a) makes review harder than necessary and (b) breaks git blame.

**Review by giordano — COMMENTED** [review]
You don't need to (or, really, shouldn't) use the wizard to update an existing recipe: https://docs.binarybuilder.org/stable/#Manually-create-or-edit-build_tarballs.jl

**Comment by n0wis** [comment]
I don't understand why the the buildkite check runner-dynamically-launch-pipelines failed.
@giordano Do you have any insight? Thanks a lot!

**Comment by giordano** [comment]
#9748

**Comment by n0wis** [comment]
Hey @giordano, by now a new version of libzenohc was released (1.0.3). Should I create a new pull request for the update? Or should I rather wait until this one is merged?

**Comment by n0wis** [comment]
Alright, let's merge this first and then I'll create a new PR without (mis)using the wizard this time.

**Pull request merged**